post_info.php

```diff
@@ -0,0 +1,695 @@
+<?php
+require 'post_query_generators.php';
+require 'db/config.php';
+// header("Content-type: application/json");
+// below is the block for receiving POST request from the frontend
+try {
+    $GLOBALS["conn"] = new PDO(
+        "mysql:host=$servername;dbname=$database",
+        $username,
+        $password
+    );
+    // Get the operations that the user wants to perform
+    $action = $_REQUEST['action'];
+    // Check the validation of $action
+    if (
+        $action != "add_or_update" &&
+        $action != "branch" &&
+        $action != "delete"
+    ) {
+        throw new Exception("'action' attribute not valid!");
+    }
+    // Decode the body of the request
+    $data = json_decode(file_get_contents('php://input'));
+    // Determine the objects that the user wants to update
+    switch ($_REQUEST['what']) {
+        case 'surveys':
+            // Get the id of the survey
+            $survey_id = $_REQUEST['survey_id'];
+            // Get the level of the survey setting, which could be "dept", "course", "section"
+            $level = $_REQUEST['level'];
+            // Check the validation of $level
+            if ($level != "dept" && $level != "course" && $level != "section") {
+                throw new Exception("'level' attribute not valid!");
+            }
+            // Get the utorid of the user who sets this survey
+            $user_id = $_REQUEST['user_id'];
+            // Call the function handle_survey_setting to deal with different situations
+            handle_survey_setting($survey_id, $level, $user_id, $action, $data);
+
+            break;
+        case 'user_info':
+            // Get the list of user info that the user wants to update
+            $user_list = $data->user_list;
+            // Call the function handle_user_info
+            handle_user_info($user_list, $action);
+            // $sql = "insert into users (user_id, name, photo) values ('admin2', 'admin2', null);";
+            break;
+        case 'course_pairings':
+            // Determine if the user wants to update the user_associations or the
+            // courses/sections of the term
+            if ($_REQUEST['mode'] == "user_associations") {
+                // Get the user association list from the request body data
+                $association_list = $data->association_list;
+                // Call the function handle_user_association
+                handle_user_association($association_list, $action);
+            } else {
+                // If the mode == "courses_sections"
+                // Get the user association list from the request body data
+                $association_list = $data->association_list;
+                // Call the function handle_courses_sections
+                handle_courses_sections($association_list, $action);
+            }
+            break;
+        default:
+            throw new Exception("'what' attribute not valid!");
```

Comment:
Make this more specific, like `'action'`

Reply:
Sure!